scripts/run_clippy.sh

```diff
@@ -1,2 +1,2 @@
 #!/bin/bash
-cargo clippy --all  -- -A clippy::while-let-loop -A clippy::too-many-arguments -A clippy::unit_arg -A clippy::if_same_then_else -A clippy::collapsible_if -A clippy::useless-let-if-seq -A clippy::map-entry -D warnings -A clippy::implicit-hasher -A clippy::ptr-arg
+cargo clippy --all  -- -A clippy::while-let-loop -A clippy::too-many-arguments -A clippy::unit_arg -A clippy::if_same_then_else -A clippy::collapsible_if -A clippy::useless-let-if-seq -A clippy::map-entry -D warnings -A clippy::implicit-hasher -A clippy::ptr-arg -A clippy::large-enum-variant -A clippy::new-ret-no-self
```

Comment:
Have you considered disabling the checks only for the specific lines of code instead of globally?

`clippy::new-ret-no-self` is a useful lint if you want to keep your code in line with the rest of the ecosystem.

`clippy::large-enum-variant` might be an indicator of possible performance problems.

The rest of the disabled checks are also worth to be fixed instead of ignoring, in my opinion.